omnigan/eval_metrics.py

```diff
@@ -63,4 +65,39 @@ def f1_score(pred_im, gt_im):
 def accuracy(pred_im, gt_im):
     pred = np.array(pred_im)
     gt = np.array(gt_im)
+    if len(pred.shape) > len(gt_im.shape):
+        pred = np.argmax(pred, axis=1)
     return float((pred == gt).sum()) / gt.size
+
+
+def mIOU(label, pred, num_classes=19):
+    """
+    https://stackoverflow.com/questions/62461379/multiclass-semantic-segmentation-model-evaluation
+    """
+    pred = F.softmax(pred, dim=1)
+    pred = torch.argmax(pred, dim=1).squeeze(1)
+    iou_list = list()
+    present_iou_list = list()
+
+    num_classes = pred.shape[-3]
+    pred = pred.view(-1)
+    label = label.view(-1)
+    # Note: Following for loop goes from 0 to (num_classes-1)
+    # and ignore_index is num_classes, thus ignore_index is
+    # not considered in computation of IoU.
+    for sem_class in range(num_classes):
+        pred_inds = pred == sem_class
+        target_inds = label == sem_class
+        if target_inds.long().sum().item() == 0:
+            iou_now = float("nan")
```

Comment:
this is not useful, is it ? since you actually don't use iou_now anywhere in this case right ?

Reply:
indeed if we remove iou_list